src/components/AddComment.jsx

```diff
@@ -0,0 +1,70 @@
+import { useState } from "react"
+import { postCommentByArticleID } from "../api";
+
+const AddComment = ({article_id, setIsListed}) => {
+
+    const [clicked, setIsClicked] = useState(false)
+    const [form, setForm] = useState({});
+    const [isListing, setIsListing] = useState(false);
+    const [disabled, setDisabled] = useState(true)
+
+    const handleClick = () => {
+      setIsClicked((prevState)=> !prevState)
+    }
+
+    const handleChange = (event) => {
+      setForm((currentForm) => {
+        const information = { ...currentForm, [event.target.id]: event.target.value }
+        const username = information.username
+        const body = information.body
+        if (username === undefined || body === undefined || body.length === 0) {
+         setDisabled(true)
+         return information
+        } 
+        else {
+         setDisabled(false)
+         return information;
+        }
+      })
+    }
+    
+    const handleSubmit = (event) => {
+      event.preventDefault()
+      setIsListing(true);
+      postCommentByArticleID(article_id, form).then(() => {
+          setIsListing(false);
+          setIsListed(true);
+      })
+    }
+
+
+
+    if (clicked===false) return <button onClick={handleClick} className="addComment">Add a Comment</button>
+    else if (isListing) return (<><h2 className="addCommentMessage">Submitting your comment... </h2><h3 className="dontRefresh">Please do not refresh the browser</h3></>);
+    else 
+    return (
+        <>
+          <button onClick={handleClick} className="addComment">Add a Comment</button>
+          <form  onSubmit={handleSubmit} id="form">
+             <h3 className= "formTitle">Add a Comment - *All fields are mandatory</h3>
+             <label htmlFor="username"  className="formSubHeading">*Username:</label>
+            <select id="username" onChange={handleChange} >
+             <option disabled selected value>
+             Select a username
+             </option>
+              <option value="jessjelly">jessjelly</option>
```

Comment:
I'd like you to have a think about how this is working from a user perspective. Imagine this were facebook - there would be millions of names on that dropdown! It also suggests that anybody can post under any username. Take a look at having a 'logged in user' value stored somewhere so that this would be more along the lines of what you would see online (bonus points if you use react context)